src/components/paster.ts

```diff
@@ -207,4 +252,390 @@ export class Paster {
 
         return text
     }
+
```

Comment:
In general, too many codes on path manipulation and file copying/moving. The whole workflow should be simplified, and the extension should have as little footprint on files as possible. In my opinion, file on disk -> use the path. Not on disk -> save to default with some tweaks for users.

Reply:
Not sure what to do to further reduce path manipulation, I think with the changes I've just made the other half of your comment is resolved.